docs/12-design.md

```diff
@@ -239,33 +243,36 @@ An outline of the design of this system is shown below:
 
 The purpose of a compaction job is to read N files and replace them with one file. This process keeps the number
 of files for a partition small, which means the number of files that need to be read in a query is small. The input
-files are all from within one partition and contain records sorted by key and sort fields.
-The output from the job is a sorted file. As the input files are sorted, it is simple to write out a sorted
-file containing their data. There are two types of compaction job: non-splitting and splitting. A non-splitting job
-is one in which the files are in a partition that is a leaf partition. In this case there is only one output file
-and it is in the same partition as the input files. A splitting job is one in which the files are in a partition
-that is not a leaf partition. In this case two output files are created, one for each of the child partitions.
-Currently, there are separate queues, ECS clusters and lambdas for these two types of jobs, although there is
-no intrinsic reason for them to be separate.
-
-When a compaction job completes, it needs to update the state store to change the status of the input files to
-ready-for-garbage-collection and it needs to add the output file(s) as active file(s). This update must be done
-atomically, to avoid clients that are requesting the state of the state store from seeing an inconsistent view.
+files contain records sorted by key and sort fields, and are filtered so that only data for the current partition is read.
+The data for an input file that exists within a specific partition can be represented by a file reference.
+The output from the job is a sorted file. As the filtered input files are sorted, it is simple to write out a sorted
+file containing their data. The output file will be written to the same partition that the input files were in.
+Note that a compaction job will only be created for files that are in leaf partitions at the time of creation, 
+meaning a partition split could happen before the job has run.
```

Comment:
It looks like a point has been removed about input files all being in the same partition, and compaction jobs only being created for leaf partitions.

Reply:
Thanks, added that bit back in